nucypher/blockchain/eth/deployers.py

```diff
@@ -505,21 +505,68 @@ class StakingEscrowDeployer(BaseContractDeployer, UpgradeableContractMixin, Owna
 
     agency = StakingEscrowAgent
     contract_name = agency.contract_name
+    contract_name_stub = "StakingEscrowStub"
 
     can_be_idle = True
-    preparation_steps = ('contract_deployment', 'dispatcher_deployment')
+    init_steps = ('stub_deployment', 'dispatcher_deployment')
+    preparation_steps = ('contract_deployment', 'dispatcher_retarget')
     activation_steps = ('approve_reward_transfer', 'initialize')
     deployment_steps = preparation_steps + activation_steps
     _proxy_deployer = DispatcherDeployer
 
-    def __init__(self, test_mode: bool = False, *args, **kwargs):
+    def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.__dispatcher_contract = None
 
         token_contract_name = NucypherTokenDeployer.contract_name
+        policy_manager_proxy_name = PolicyManagerDeployer._proxy_deployer.contract_name
+        policy_manager_contract_name = PolicyManagerDeployer.contract_name
+        adjudicator_proxy_name = AdjudicatorDeployer._proxy_deployer.contract_name
+        adjudicator_contract_name = AdjudicatorDeployer.contract_name
+        worklock_contract_name = WorklockDeployer.contract_name
         self.token_contract = self.blockchain.get_contract_by_name(registry=self.registry,
                                                                    contract_name=token_contract_name)
-        self.test_mode = test_mode
+        try:
+            self.policy_manager = self.blockchain.get_contract_by_name(registry=self.registry,
+                                                                       contract_name=policy_manager_contract_name,
+                                                                       proxy_name=policy_manager_proxy_name)
+        except self.registry.UnknownContract:
+            self.policy_manager = None
```

Comment:
Why allow agents be `None` instead of enforcing them to be enrolled?

Reply:
It's not agents, just contracts which will be not deployed yet at stub deployment step